### PR TITLE
UI: Autoscroll to Error Field

### LINKED ISSUE #4798: GSoC 2021 Idea - UX improvements in the UI
## Background

Surely our new UI is great, but some users might find it interesting to navigate with shortcuts and kayboard only

## Proposed Tasks

This could be interesting to any UI/UX engineer, not required huge domain knowledge which usually turns down candidates.
 

1. Being able to navigate without mouse and use shortkey – for example when on the dashboard when the user presses T, he could be directly transitioned to Templates. Or I for infrastructure, G – global settings, H for hosts etc.. While in networks/instances/volumes press A – add item(network,volume etc). Same goes for all buttons inside a page like Instances (stop, start, destroy etc)
Screenshot 2021-03-11 at 14 57 32
Screenshot 2021-03-11 at 14 53 19

List of suggestions of shortcuts: 
- [i] - infrastructure
- [g] - configuration->global settings
- [v] - storage->volumes
- [c] - compute->instances
- [t] - images->templates
- [n] - networks 
- [a] - accounts
- [p] - projects
- [e] - events
_Please bring up this list to community mailing lists for comments and objections. Also someone could bring up more ideas_

2. Confirm dialogues with key – This is not working consistently, some dialogue windows have it some do not. But surely it’ll be great if we can confirm by pressing Enter or Space and cancel by Esc.
 
3. Forms submission improvements – For example, when deploying instance if you simply click ok without selected network it will prompt you with error and then you’ll need to scroll back to the network place and add it. We could be more intuitive and automatically move the focus of the user back at the networks place in the form, where user can easily add or select a network.
 
4. Placement of buttons and forms – I’ve pointed this as debatable since it’s more of a UX discussion, which may end up with a decision ‘there is nothing to fix’. The idea is to think about questions as is this form intuitive enough, or is this button placed good, maybe there’s a more suitable place for it. Etc.

### Relevant Skills
- Vue.js

## Proposed Mentor

Boris Stoyanov - requirements and ideas
David Jumani - coding part
Any developer from the community - Coding issues 

## Difficulty

Easy

## Potential Mentors

Boris Stoyanov - requirements and ideas (boris.stoyanov@shapepblue.com)
David Jumani - coding part (david.jumani@shapeblue.com)
Rohit Yadav - any help/guidance (rohit@apache.org)

##### ISSUE TYPE

 * Feature Idea

### PR DESCRIPTION
**Pull request by apurv-1**
### Description
fixes #4798 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### Screenshots (if appropriate):

https://user-images.githubusercontent.com/53378650/131230082-358ac204-4925-43da-8d26-bfbe66cbd76e.mp4

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by apurv-1** [comment]
@blueorangutan ui

**Comment by sureshanaparti** [comment]
Hi @apurv-1 can you please fix the conflicts. thanks.

**Comment by apurv-1** [comment]
Done @sureshanaparti 

> Hi @apurv-1 can you please fix the conflicts. thanks.

File: ui/src/views/compute/DeployVM.vue [hunk 1]
```diff
@@ -1419,7 +1424,7 @@ export default {
             })
             return
           }
-
+          // console.log(err.zoneid.errors[0])
```
Comment (DaanHoogland) [inline ui/src/views/compute/DeployVM.vue:1427]: let's not put code in comments. we have git to stash or archive code.

Comment (apurv-1) [reply ui/src/views/compute/DeployVM.vue:1427]: Yeah, sorry will fix it asap!

**Comment by apurv-1** [comment]
@blueorangutan ui

**Comment by apurv-1** [comment]
@blueorangutan ui

**Review by borisstoyanov — APPROVED** [review]
LGTM

**Review by shwstppr — APPROVED** [review]

**Review by utchoang — APPROVED** [review]

**Comment by rhtyd** [comment]
Merged this as it appears this has been already tested by @borisstoyanov and others.

**Pull request merged**